src/database/functions/GuildSettingsFunctions.ts

```diff
@@ -63,90 +64,165 @@ export async function removeGuildSettings(guild_id: string) {
 }
 
 export async function UpdateWelcome(
-    guildID:string,
-    data:{
-        enable?:boolean,
-        welcomeDM?: boolean,
-        welcomeChannel?: boolean,
-        welcomeChannelID?: string,
-        channelMessage?: string,
-        dmMessage?: string,
-        welcomeRoles?: Array<string>,
-    }){
-        let guildData:rawGuildSettings | null = await guildSettings.findOne({guild_id:guildID})
-        if(!guildData)return
-        let Settings = guildData?.welcomeSettings
-        if(data.enable == undefined){data.enable = Settings.enable}
-        if(data.welcomeDM == undefined){data.welcomeDM = Settings.welcomeDM}
-        if(data.welcomeChannel == undefined){data.welcomeChannel = Settings.welcomeChannel}
-        if(data.welcomeChannelID == undefined){data.welcomeChannelID = Settings.welcomeChannelID}
-        if(data.channelMessage == undefined){data.channelMessage = Settings.channelMessage}
-        if(data.dmMessage == undefined){data.dmMessage = Settings.dmMessage}
-        if(data.welcomeRoles == undefined){data.welcomeRoles = Settings.welcomeRoles}
-
-        await guildSettings.updateOne({guild_id:guildID},{welcomeSettings:data})
+    guildID: string,
+    data: {
+        enable?: boolean;
+        welcomeDM?: boolean;
+        welcomeChannel?: boolean;
+        welcomeChannelID?: string;
+        channelMessage?: string;
+        dmMessage?: string;
+        welcomeRoles?: Array<string>;
+        CustomWelcomeBackground?: string;
+    }
+) {
+    let guildData: rawGuildSettings | null = await guildSettings.findOne({
+        guild_id: guildID,
+    });
+    if (!guildData) return;
+    let Settings = guildData?.welcomeSettings;
+    if (data.enable == undefined) {
+        data.enable = Settings.enable;
+    }
+    if (data.welcomeDM == undefined) {
+        data.welcomeDM = Settings.welcomeDM;
+    }
+    if (data.welcomeChannel == undefined) {
+        data.welcomeChannel = Settings.welcomeChannel;
+    }
+    if (data.welcomeChannelID == undefined) {
+        data.welcomeChannelID = Settings.welcomeChannelID;
+    }
+    if (data.channelMessage == undefined) {
```

Comment:
```suggestion
    if (data.welcomeChannelMessage == undefined) {
```

Reply:
If(!data.something)

Reply:
no some items are bools and it will mess with it (experience first hand)